commerce-headless-graphql/src/main/java/com/liferay/commerce/headless/graphql/internal/servlet/CommerceGraphQLServlet.java

```diff
@@ -0,0 +1,295 @@
+
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.GraphQLError;
+import graphql.GraphqlErrorHelper;
+
+import graphql.introspection.IntrospectionQuery;
+
+import graphql.schema.idl.RuntimeWiring;
+import graphql.schema.idl.SchemaGenerator;
+import graphql.schema.idl.SchemaParser;
+import graphql.schema.idl.TypeDefinitionRegistry;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import java.nio.charset.Charset;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.apache.commons.io.IOUtils;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Marco Leo
+ */
+@Component(
+	immediate = true,
+	property = {
+		"osgi.http.whiteboard.context.path=/test",
+		"osgi.http.whiteboard.servlet.name=com.liferay.commerce.headless.graphql.internal.servlet.CommerceGraphQLServlet",
+		"osgi.http.whiteboard.servlet.pattern=/test/*"
+	},
+	service = Servlet.class
+)
+public class CommerceGraphQLServlet extends HttpServlet {
+
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_graphQL = _buildGraphQL();
+
+		_objectMapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
+		_objectMapper.registerModule(new Jdk8Module());
+	}
+
+	@Override
+	protected void doGet(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws IOException, ServletException {
+
+		ExecutionResult executionResult = _graphQL.execute(
+			IntrospectionQuery.INTROSPECTION_QUERY);
+
+		String json = _toString(executionResult);
+
+		httpServletResponse.setContentType(ContentTypes.APPLICATION_JSON);
+
+		ServletResponseUtil.write(httpServletResponse, json);
+	}
+
+	@Override
+	protected void doPost(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws IOException, ServletException {
+
+		try {
+			HttpSession httpSession = httpServletRequest.getSession();
+
+			if (PortalSessionThreadLocal.getHttpSession() == null) {
+				PortalSessionThreadLocal.setHttpSession(httpSession);
+			}
+
+			long companyId = _portal.getCompanyId(httpServletRequest);
+
+			User user = _portal.getUser(httpServletRequest);
+
+			if (user == null) {
+				user = _userLocalService.getDefaultUser(companyId);
+			}
+
+			PermissionChecker permissionChecker =
+				PermissionCheckerFactoryUtil.create(user);
+
+			PermissionThreadLocal.setPermissionChecker(permissionChecker);
+
+			String body = IOUtils.toString(
```

Comment:
We should be able to use `StringUtil.read` for this. The only differences are (1) it decodes the string into UTF-8 rather than the system default charset, (2) it standardizes line endings to all be `\n`, and (3) it trims the result. If those are acceptable differences, we shouldn't need this dependency at all.